xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -414,3 +494,946 @@ func (s) TestEDSResourceRemoved(t *testing.T) {
+					"cluster": "%s"
+				}
+			}]
+		}`, clusterName)
+	scpr := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(jsonSC)
+	r.InitialState(xdsclient.SetClient(resolver.State{ServiceConfig: scpr}, xdsClient))
+
+	// Create a ClientConn and make a successful RPC.
+	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
+	if err != nil {
+		t.Fatalf("Failed to dial local test server: %v", err)
+	}
+	defer cc.Close()
+
+	client := testgrpc.NewTestServiceClient(cc)
+	peer := &peer.Peer{}
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(peer)); err != nil {
+		t.Fatalf("EmptyCall() RPC failed: %v", err)
+	}
+	if peer.Addr.String() != addrs[0].Addr {
+		t.Fatalf("EmptyCall() RPC routed to backend %q, want %q", peer.Addr, addrs[0].Addr)
+	}
+
+	// Ensure EDS watch is registered for eds_service_name.
+	select {
+	case <-ctx.Done():
+		t.Fatal("Timeout when waiting for EDS request to be received on the management server")
+	case names := <-edsResourceNameCh:
+		if !cmp.Equal(names, []string{edsServiceName}) {
+			t.Fatalf("Received EDS request with resource names %v, want %v", names, []string{edsServiceName})
+		}
+	}
+
+	// Change the cluster resource to not contain an eds_service_name.
+	resources.Clusters = []*v3clusterpb.Cluster{e2e.DefaultCluster(clusterName, "", e2e.SecurityLevelNone)}
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure that an EDS watch for eds_service_name is canceled and new watch
+	// for cluster_name is registered. The actual order in which this happens is
+	// not deterministic, i.e the watch for old resource could be canceled
+	// before the new one is registered or vice-versa. In either case,
+	// eventually, we want to see a request to the management server for just
+	// the cluster_name.
+	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
+		names := <-edsResourceNameCh
+		if cmp.Equal(names, []string{clusterName}) {
+			break
+		}
+	}
+	if ctx.Err() != nil {
+		t.Fatalf("Timeout when waiting for old EDS watch to be canceled and new one to be registered")
+	}
+
+	// Make RPC, and ensure that it gets routed to the second backend.
+	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
+		if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(peer)); err != nil {
+			continue
+		}
+		if peer.Addr.String() == addrs[1].Addr {
+			break
+		}
+	}
+	if ctx.Err() != nil {
+		t.Fatal("Timeout when waiting for EmptyCall() RPC routed to correct backend")
+	}
+
+	// Change cluster resource circuit breaking count.
+	resources.Clusters[0].CircuitBreakers = &v3clusterpb.CircuitBreakers{
+		Thresholds: []*v3clusterpb.CircuitBreakers_Thresholds{
+			{
+				Priority:    v3corepb.RoutingPriority_DEFAULT,
+				MaxRequests: wrapperspb.UInt32(512),
+			},
+		},
+	}
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
```

Comment:
I used to see these errors wrapped with "error updating management server: %v". Did we add this contextual information language to the managementServer to prevent needing to log this? I see this error as a possible error returned from this Update call: return fmt.Errorf("failed to create new snapshot cache: %v", err). I feel like reading that as is gives no context on what thing/reason emitted the error. Should we add this language here in the test and elsewhere in our codebase to provide more context?

Reply:
I just meant the fact the error was emitted from the management server. Reading that error as is would give me little context as to what component is failing. Although, I see your point about this never failing. I've never seen it fail myself personally. Up to you whether you want to specify component which failed. I feel like we had that information everywhere previously, but that doesn't seem to be the case anymore, so I'm fine with leaving this as is.